core/src/main/java/io/confluent/rest/RestConfig.java

```diff
@@ -33,9 +33,16 @@
   protected static final boolean DEBUG_CONFIG_DEFAULT = false;
 
   public static final String PORT_CONFIG = "port";
-  protected static final String PORT_CONFIG_DOC = "Port to listen on for new connections.";
+  protected static final String PORT_CONFIG_DOC = "DEPRECATED: port to listen on for new HTTP connections. Use " +
+      "listeners instead.";
   protected static final int PORT_CONFIG_DEFAULT = 8080;
 
+  public static final String LISTENERS_CONFIG = "listeners";
+  protected static final String LISTENERS_DOC = "List of listeners. http and https are supported. Each " +
+      "listener must include the protocol, hostname, and port. For example: http://myhost:8080," +
+      "https://0.0.0.0:8081";
+  protected static final String LISTENERS_DEFAULT = ""; // TODO: add a default value when `PORT_CONFIG` is deprecated.
```

Comment:
Yes, "deleted" is correct, not "deprecated."

As for your alternative approach, I think this will complicate the code, because I'll need to check if a port is configured, and if so, use that port instead of the port specified in the default listener. Unless I'm misunderstanding your suggestion? If not, I prefer my current approach. But thanks for your suggestion.